"index out of range" cause ExecuteResponse.Results were empty.

Got an error:
```
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/Boostport/avatica.(*stmt).exec(0xc4201e80f0, 0x8f2ee0, 0xc4200160a0, 0xc4201e81e0, 0x2, 0x2, 0x0, 0x0, 0x709460, 0x738800)
        /home/yura/go/src/github.com/Boostport/avatica/statement.go:81 +0x2c3
github.com/Boostport/avatica.(*stmt).ExecContext(0xc4201e80f0, 0x8f2ee0, 0xc4200160a0, 0xc4201e8140, 0x2, 0x2, 0xc400000008, 0xc420399d00, 0x0, 0x4c8b06)
        /home/yura/go/src/github.com/Boostport/avatica/statement_go18.go:19 +0x117
database/sql.ctxDriverStmtExec(0x8f2ee0, 0xc4200160a0, 0x8f3120, 0xc4201e80f0, 0xc4201e8140, 0x2, 0x2, 0x2, 0x2, 0x0, ...)
        /snap/go/current/src/database/sql/ctxutil.go:66 +0x22e
database/sql.resultFromStatement(0x8f2ee0, 0xc4200160a0, 0x8f1960, 0xc420110120, 0xc420281b08, 0xc420281eb0, 0x2, 0x2, 0x0, 0x0, ...)
        /snap/go/current/src/database/sql/sql.go:2109 +0x184
database/sql.(*DB).execDC(0xc42008a960, 0x8f2ee0, 0xc4200160a0, 0xc420134070, 0xc420281c60, 0x77b0de, 0xd5, 0xc420281eb0, 0x2, 0x2, ...)
        /snap/go/current/src/database/sql/sql.go:1272 +0x2c7
database/sql.(*DB).exec(0xc42008a960, 0x8f2ee0, 0xc4200160a0, 0x77b0de, 0xd5, 0xc420281eb0, 0x2, 0x2, 0xc420016101, 0x6eb580, ...)
```
Looks like Results were empty in this row:
```go
    // Currently there is only 1 ResultSet per response
    changed := int64(res.(*message.ExecuteResponse).Results[0].UpdateCount)                                                                                                                                                         
```
Sorry, I don't know how reliably reproduce it. I was copying large table to other table (using `upsert into t2 select some from t1`), and it happens at night. Probably it is due to whole cluster instability (other clients died with "cluster is out of memory" probably at same time).

--- a/rows.go
+++ b/rows.go
@@ -51,7 +51,7 @@ type rows struct {
 // string should be returned for that entry.
 func (r *rows) Columns() []string {
 
-	cols := []string{}
+	var cols []string
 
 	for _, column := range r.resultSets[r.currentResultSet].columns {
 		cols = append(cols, column.name)
@@ -101,7 +101,7 @@ func (r *rows) Next(dest []driver.Value) error {
 
 		frame := res.(*message.FetchResponse).Frame
 
-		data := [][]*message.TypedValue{}
+		var data [][]*message.TypedValue
 
 		// In some cases the server does not return done as true
 		// until it returns a result with no rows
@@ -110,7 +110,7 @@ func (r *rows) Next(dest []driver.Value) error {
 		}
 
 		for _, row := range frame.Rows {
-			rowData := []*message.TypedValue{}
+			var rowData []*message.TypedValue
 
 			for _, col := range row.Value {
 				rowData = append(rowData, col.ScalarValue)
@@ -137,13 +137,14 @@ func (r *rows) Next(dest []driver.Value) error {
 // newRows create a new set of rows from a result set.
 func newRows(conn *conn, statementID uint32, resultSets []*message.ResultSetResponse) *rows {
 
-	rsets := []*resultSet{}
+	var rsets []*resultSet
 
 	for _, result := range resultSets {
 		if result.Signature == nil {
 			break
 		}
-		columns := []*column{}
+
+		var columns []*column
 
 		for _, col := range result.Signature.Columns {
 
@@ -226,10 +227,10 @@ func newRows(conn *conn, statementID uint32, resultSets []*message.ResultSetResp
 
 		frame := result.FirstFrame
 
-		data := [][]*message.TypedValue{}
+		var data [][]*message.TypedValue
 
 		for _, row := range frame.Rows {
-			rowData := []*message.TypedValue{}
+			var rowData []*message.TypedValue
 
 			for _, col := range row.Value {
 				rowData = append(rowData, col.ScalarValue)

--- a/statement.go
+++ b/statement.go
@@ -6,9 +6,10 @@ import (
 
 	"math"
 
+	"errors"
+
 	"github.com/Boostport/avatica/message"
 	"golang.org/x/net/context"
-	"errors"
 )
 
 type stmt struct {
@@ -80,7 +81,7 @@ func (s *stmt) exec(ctx context.Context, args []namedValue) (driver.Result, erro
 
 	results := res.(*message.ExecuteResponse).Results
 
-	if len(results) <= 0{
+	if len(results) <= 0 {
 		return nil, errors.New("empty ResultSet in ExecuteResponse")
 	}
 
@@ -130,7 +131,7 @@ func (s *stmt) query(ctx context.Context, args []namedValue) (driver.Rows, error
 
 func (s *stmt) parametersToTypedValues(vals []namedValue) []*message.TypedValue {
 
-	result := []*message.TypedValue{}
+	var result []*message.TypedValue
 
 	for i, val := range vals {
 		typed := message.TypedValue{}

--- a/statement.go
+++ b/statement.go
@@ -8,6 +8,7 @@ import (
 
 	"github.com/Boostport/avatica/message"
 	"golang.org/x/net/context"
+	"errors"
 )
 
 type stmt struct {
@@ -77,8 +78,14 @@ func (s *stmt) exec(ctx context.Context, args []namedValue) (driver.Result, erro
 		return nil, err
 	}
 
+	results := res.(*message.ExecuteResponse).Results
+
+	if len(results) <= 0{
+		return nil, errors.New("empty ResultSet in ExecuteResponse")
+	}
+
 	// Currently there is only 1 ResultSet per response
-	changed := int64(res.(*message.ExecuteResponse).Results[0].UpdateCount)
+	changed := int64(results[0].UpdateCount)
 
 	return &result{
 		affectedRows: changed,

